File: app/src/main/java/vn/loitp/app/activity/demo/firebase/auth/AuthFirebaseMenuActivity.java
package vn.loitp.app.activity.demo.firebase.auth;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import loitp.basemaster.R;
import vn.loitp.core.base.BaseFontActivity;
import vn.loitp.core.utilities.LActivityUtil;

//https://github.com/firebase/quickstart-android
public class AuthFirebaseMenuActivity extends BaseFontActivity implements View.OnClickListener {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        findViewById(R.id.bt_gg).setOnClickListener(this);
        findViewById(R.id.bt_fb).setOnClickListener(this);
    }

    @Override
    protected boolean setFullScreen() {
        return false;
    }

    @Override
    protected String setTag() {
        return getClass().getSimpleName();
    }

    @Override
    protected int setLayoutResourceId() {
        return R.layout.activity_auth_firebase;
    }

    @Override
    public void onClick(View v) {
        Intent intent = null;
        switch (v.getId()) {
            case R.id.bt_gg:
                intent = new Intent(activity, AuthFirebaseGoogleActivity.class);
                break;
            case R.id.bt_fb:
                intent = new Intent(activity, AuthFirebaseFacebookActivity.class);
                break;
        }
        if (intent != null) {
            startActivity(intent);
            LActivityUtil.INSTANCE.tranIn(activity);
        }
    }
}

File: app/src/main/java/vn/loitp/app/activity/demo/firebase/auth/AuthFirebaseMenuActivity.kt
package vn.loitp.app.activity.demo.firebase.auth

import android.content.Intent
import android.os.Bundle
import android.view.View
import com.annotation.IsFullScreen
import com.annotation.LogTag

import com.core.base.BaseFontActivity
import com.core.utilities.LActivityUtil
import kotlinx.android.synthetic.main.activity_auth_firebase.*

import vn.loitp.app.R

//https://github.com/firebase/quickstart-android

@LogTag("AuthFirebaseMenuActivity")
@IsFullScreen(false)
class AuthFirebaseMenuActivity : BaseFontActivity(), View.OnClickListener {

    override fun setLayoutResourceId(): Int {
        return R.layout.activity_auth_firebase
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        btGG.setOnClickListener(this)
        btFb.setOnClickListener(this)
    }

    override fun onClick(v: View) {
        var intent: Intent? = null
        when (v) {
            btGG ->
                intent = Intent(this, AuthFirebaseGoogleActivity::class.java)
            btFb ->
                intent = Intent(this, AuthFirebaseFacebookActivity::class.java)
        }
        intent?.let {
            startActivity(it)
            LActivityUtil.tranIn(this)
        }
    }
}
